### PR TITLE
poc: enable doc-value-only mappings for APM data streams

### PR DESCRIPTION
**Pull request by kruskall**
## Motivation/summary

Mark `timestamp.us` as doc-value-only field for a quick poc.

Add test to verify the field is indexed as expected.

This is a PoC and `timestamp.us` is used as an example. The actual list of fields should be decided after coordinating with the UI team.

## Checklist

- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)
- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)
- [ ] Documentation has been updated

For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

## Related issues

Related to https://github.com/elastic/apm-server/issues/9183

### CODE REVIEW

File: apmpackage/apm/data_stream/traces/manifest.yml [hunk 1]
```diff
@@ -16,3 +16,9 @@ elasticsearch:
             mapping:
               type: scaled_float
               scaling_factor: 1000000
+      properties:
+        timestamp:
+          properties:
+            us:
+              type: long
+              index: false
```
Comment (axw) [inline apmpackage/apm/data_stream/traces/manifest.yml:24]: I think we should instead update `fields/fields.yml`:

https://github.com/elastic/apm-server/blob/ff5e1792298a52fffef1e5079917942b6b364c94/apmpackage/apm/data_stream/traces/fields/fields.yml#L248-L251

IIANM you can just add `index: false` to the field definition there.

**Review by axw — CHANGES_REQUESTED** [review]
I'm not convinced that the test is necessary, since it's second guessing that Fleet is working as expected. WDYT?

File: apmpackage/apm/data_stream/traces/manifest.yml (hunk 1 above)
Comment (kruskall) [reply apmpackage/apm/data_stream/traces/manifest.yml]: Thanks for the tip! :bow: 

I've updated the poc to use the fields mentioned in the related issue

**Comment by kruskall** [comment]
> I'm not convinced that the test is necessary, since it's second guessing that Fleet is working as expected. WDYT?

Agree, I've removed the test. I've updated the PR to use the new fields.

File: apmpackage/apm/data_stream/traces/manifest.yml (hunk 1 above)
Comment (axw) [reply apmpackage/apm/data_stream/traces/manifest.yml]: Thanks, looks good :)

**Review by axw — APPROVED** [review]
Looks good!

Later on we should revisit _how_ we define the fields that should or should not be indexed. For example we might update the package generator (apmpackage/cmd/genpackage) to inject the property across all the data streams for certain fields.

That would save us from having to manually keep them in sync. Also it would enable us to define some policies, like "all metric fields should be doc-value-only". We might also want to opt into indexing, rather than opting out, depending on how many fields are actually used in searches.

**Pull request merged**